### PR TITLE
CP-25990: Enable Create VLAN on SR-IOV network

### PR DESCRIPTION
**Pull request by jishengx**
Signed-off-by: Jisheng Xing <jisheng.xing@citrix.com>

### CODE REVIEW

**Comment by cheng-z** [comment]
Hi @MihaelaStoica 
Could you please help to have a look of this PR?
Thanks.

File: XenAdmin/Wizards/NewNetworkWizard.cs [hunk 1]
```diff
@@ -227,7 +227,17 @@ private void CreateSRIOV()
         private void CreateNonBonded()
         {
             XenAPI.Network network = PopulateNewNetworkObj();
-            PIF nic = pageNetworkDetails.SelectedHostNic;
+
+            PIF nic;
+            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            {
+                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];
+                nic = Pool != null ? Pool.Connection.Resolve(Pool.Connection.Resolve(sriovLogicalPif).logical_PIF) : Host.Connection.Resolve(Host.Connection.Resolve(sriovLogicalPif).logical_PIF);
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewNetworkWizard.cs]: Can use xenConnection here. This won't be null.
Also to avoid nested Resolve calls here, we can set the sriovLogicalPif variable (should it be called sriovPhysicalPif?) to the resolved PIF object, on line 234.

File: XenAdmin/Wizards/NewNetworkWizard.cs [hunk 2]
```diff
@@ -227,7 +227,17 @@ private void CreateSRIOV()
         private void CreateNonBonded()
         {
             XenAPI.Network network = PopulateNewNetworkObj();
-            PIF nic = pageNetworkDetails.SelectedHostNic;
+
+            PIF nic;
+            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            {
+                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewNetworkWizard.cs]: To be on the safe side, we should check that pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of.Count > 0

File: XenAdmin/Wizards/NewNetworkWizard_Pages/NetWDetails.cs [hunk 3]
```diff
@@ -139,6 +144,7 @@ private void UpdateEnablement(bool external, Host host)
             labelNIC.Visible = external;
             if (comboBoxNICList.Items.Count > 0)
                 comboBoxNICList.SelectedIndex = external ? comboBoxNICList.Items.Count - 1 : -1;
+            groupBoxSriovConfig.Visible = SelectedHostNic.sriov_physical_PIF_of.Count != 0;
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewNetworkWizard_Pages/NetWDetails.cs]: SelectedHostNic can be null, for example when comboBoxNICList is not visible

File: XenAdmin/Wizards/NewNetworkWizard.cs [hunk 4]
```diff
@@ -229,15 +229,25 @@ private void CreateNonBonded()
             XenAPI.Network network = PopulateNewNetworkObj();
 
             PIF nic;
-            if(pageNetworkDetails.createVlanOnSriovNetwork)
+            if (pageNetworkDetails.CreateVlanOnSriovNetwork)
             {
-                var sriovLogicalPif = pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of[0];
-                nic = Pool != null ? Pool.Connection.Resolve(Pool.Connection.Resolve(sriovLogicalPif).logical_PIF) : Host.Connection.Resolve(Host.Connection.Resolve(sriovLogicalPif).logical_PIF);
+                if (pageNetworkDetails.SelectedHostNic == null || pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of == null || pageNetworkDetails.SelectedHostNic.sriov_physical_PIF_of.Count == 0)
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewNetworkWizard.cs]: Here we should use the IsSriovPhysicalPIF() function introduced in PR #1982. We can do it in a separate PR after that one is merged.

File: XenAdmin/Wizards/NewNetworkWizard_Pages/NetWDetails.cs [hunk 5]
```diff
@@ -144,7 +144,7 @@ private void UpdateEnablement(bool external, Host host)
             labelNIC.Visible = external;
             if (comboBoxNICList.Items.Count > 0)
                 comboBoxNICList.SelectedIndex = external ? comboBoxNICList.Items.Count - 1 : -1;
-            groupBoxSriovConfig.Visible = SelectedHostNic.sriov_physical_PIF_of.Count != 0;
+            groupBoxSriovConfig.Visible = SelectedHostNic != null && SelectedHostNic.sriov_physical_PIF_of != null && SelectedHostNic.sriov_physical_PIF_of.Count != 0;
```
Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewNetworkWizard_Pages/NetWDetails.cs]: Here we should use the IsSriovPhysicalPIF() function introduced in PR #1982. We can do it in a separate PR after that one is merged.
